### DISCUSSION TITLE
Tamagachi project : Modal form does not appear

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#208] [Help Me!] [answered]
Matthew Gonzales

Have you searched the other help tickets? 
Yes, but this is something different I believe.

Describe your issue: 
I'm updating old projects and trying to add a modal to my project.
I made a modal div and form and button. The button populates the modal form but it does not appear (I can see it populate in Dev Tools Console)

List any error messages that appear. 
None

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
https://stackoverflow.com/questions/16093568/bootstrap-modal-not-displaying
https://stackoverflow.com/questions/46652210/bootstrap-modal-not-showing-up-on-click
Other links don't seem helpful: either mention jQuery which I'm not using, or older versions of Bootstrap

What have you tried? Tell us everything you have tried.
I compared the way I'm generating my modal to our latest group project and also Almost Amazon and I am making it the same way.
I've double-checked all my div Ids are correct
I've made sure to have data-toggle and data-target on my button which shows the form.
I've looked in my CSS for anything which could be hiding the modal but there is nothing.

Links to src code:

appBuilder() makes initial divs:
https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/components/appBuilder.js

formModal() which creates the modal container:
https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/components/formModal.js

buttonEvents() where eventListeners live:
https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/helpers/buttonEvents.js 

petForm() which should appear after button is clicked:
https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/components/petForm.js

Thanks,

Matthew

**Comment by drteresavasquez** [comment, marked as answer]
@GonzalesMatthew so, what is happening here is you have the target to open the Modal on the DOM on the click of the button, but the modal does not exist on the DOM until the button is clicked. The modal needs to exist before the click happens so that it can open. You can move your `formModal()` function outside of the button click.

So, moving this line: https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/helpers/buttonEvents.js#L19

to here may work: https://github.com/GonzalesMatthew/ASSIGNMENT-Tamagotchi/blob/main/src/javascripts/main.js#L5

**Comment by drteresavasquez** [comment]
@GonzalesMatthew Did this solve your problem?

**Reply by GonzalesMatthew** [reply]
Just seeing this. I opted for using a form because I had no clue what was happening.
